src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1311,17 +1310,20 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 				];
 			}
 		} else {
-			// TODO: Read current args, only enable if they are recognized (ie. [] and ["-l"]), warn otherwise
-			switch (shell) {
-				case 'bash':
-					newArgs = ['--init-file', '${execInstallFolder}/out/vs/workbench/contrib/terminal/browser/media/ShellIntegration-bash.sh'];
-					break;
-				case 'pwsh':
-					newArgs = ['-noexit', '-command', '. "${execInstallFolder}/out/vs/workbench/contrib/terminal/browser/media/shellIntegration.ps1"'];
-					break;
-				case 'zsh':
-					newArgs = ['-c', '"${execInstallFolder}/out/vs/workbench/contrib/terminal/browser/media/ShellIntegration-zsh.sh"; zsh -il'];
-					break;
+			if (!originalArgs || originalArgs === [] || originalArgs === ['-l']) {
```

Comment:
I think we want to pass on the `-l` if we can, otherwise warn. So for zsh which already uses `-il`, we would use `'-i'` + (originalArgs === ['-l'] ? 'l' : '')` For pwsh I don't think `-l` is valid, etc.

Reply:
pwsh is handled above that statement and only enabled if there are no args.

Reply:
isn't it doing the same thing to concatenate it like that?

Reply:
here is my understanding of what the code does. what should it be doing instead?
```
if pwsh && args undefined
    enable and use the correct args
if bash/zsh && args undefined | [] or ['-l']
   enable and use the correct args
```

Reply:
or pwsh `-NoLogo` or `-nol`

Reply:
ignore the above bc implied